### PR TITLE
Pass namespace through RunInference transform

### PR DESCRIPTION
**Pull request by AnandInguva**
Fixes https://github.com/apache/beam/issues/21721

1. Pass a custom namespace through the RunInference transform. If none is passed, the namespace defined in the ModelHandler will be used.

------------------------

Thank you for your contribution! Follow this checklist to help us incorporate your contribution quickly and easily:

 - [ ] [**Choose reviewer(s)**](https://beam.apache.org/contribute/#make-your-change) and mention them in a comment (`R: @username`).
 - [ ] Mention the appropriate issue in your description (for example: `addresses #123`), if applicable. This will automatically add a link to the pull request in the issue. If you would like the issue to automatically close on merging the pull request, comment `fixes #<ISSUE NUMBER>` instead.
 - [ ] Update `CHANGES.md` with noteworthy changes.
 - [ ] If this contribution is large, please file an Apache [Individual Contributor License Agreement](https://www.apache.org/licenses/icla.pdf).

See the [Contributor Guide](https://beam.apache.org/contribute) for more tips on [how to make review process smoother](https://beam.apache.org/contribute/get-started-contributing/#make-the-reviewers-job-easier).

To check the build health, please visit [https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md](https://github.com/apache/beam/blob/master/.test-infra/BUILD_STATUS.md)

GitHub Actions Tests Status (on master branch)
------------------------------------------------------------------------------------------------
[Build python source distribution and wheels](https://github.com/apache/beam/actions?query=workflow%3A%22Build+python+source+distribution+and+wheels%22+branch%3Amaster+event%3Aschedule)
[Python tests](https://github.com/apache/beam/actions?query=workflow%3A%22Python+Tests%22+branch%3Amaster+event%3Aschedule)
[Java tests](https://github.com/apache/beam/actions?query=workflow%3A%22Java+Tests%22+branch%3Amaster+event%3Aschedule)
[Go tests](https://github.com/apache/beam/actions?query=workflow%3A%22Go+tests%22+branch%3Amaster+event%3Aschedule)

See [CI.md](https://github.com/apache/beam/blob/master/CI.md) for more information about GitHub Actions CI.

### CODE REVIEW

**Comment by AnandInguva** [comment]
Run Inference Benchmarks

**Comment by AnandInguva** [comment]
Run Inference Benchmarks

**Comment by AnandInguva** [comment]
Run Inference Benchmarks

**Comment by AnandInguva** [comment]
Run Inference Benchmarks

**Comment by AnandInguva** [comment]
Run Inference Benchmarks

File: sdks/python/apache_beam/ml/inference/base_test.py [hunk 1]
```diff
@@ -142,6 +142,25 @@ def test_run_inference_impl_inference_args(self):
           inference_args=inference_args)
       assert_that(actual, equal_to(examples), label='assert:inferences')
 
+  def test_run_inference_metrisc_with_custom_namespace(self):
```
Comment (AnandInguva) [inline sdks/python/apache_beam/ml/inference/base_test.py:145]:
```suggestion
  def test_run_inference_metrics_with_custom_namespace(self):
```

**Comment by AnandInguva** [comment]
R : @damccorm @ryanthompson591

**Comment by AnandInguva** [comment]
@damccorm are you able to re-run github actions on this PR?

**Comment by AnandInguva** [comment]
retest this please!

**Comment by damccorm** [comment]
> @damccorm are you able to re-run github actions on this PR?

Yes, but it won't help - you need to merge in master to pick up https://github.com/apache/beam/pull/23226 which fixes this issue

**Comment by AnandInguva** [comment]
> > @damccorm are you able to re-run github actions on this PR?
> 
> Yes, but it won't help - you need to merge in master to pick up #23226 which fixes this issue

I thought GH actions would merge master into the PR before the checks. Merged master.

File: sdks/python/apache_beam/ml/inference/base.py [hunk 2]
```diff
@@ -321,9 +324,10 @@ def expand(
         # TODO(https://github.com/apache/beam/issues/21440): Hook into the
         # batching DoFn APIs.
         | beam.BatchElements(**self._model_handler.batch_elements_kwargs())
-        | (
+        | 'RunInference_DoFn' >> (
```
[the remark this answers is not in the document]
Comment (AnandInguva) [reply sdks/python/apache_beam/ml/inference/base.py:327]: Hi, yes I thought of doing that in a different PR but I can change here as well.

[the remark this answers is not in the document]
Comment (damccorm) [reply sdks/python/apache_beam/ml/inference/base.py:327]: If you'd prefer to follow up with those in another PR that's fine (your call), I'd just ask that this particular suggested change be made while we're here.

**Comment by AnandInguva** [comment]
Run Inference Benchmarks

**Comment by damccorm** [comment]
I'll merge once precommits pass

**Pull request merged**